src/test/java/org/jabref/logic/importer/fetcher/GrobidCitationFetcherTest.java

```diff
@@ -0,0 +1,206 @@
+      Method parseUsingGrobid = GrobidCitationFetcher.class.getDeclaredMethod("parseUsingGrobid", String.class);
+      parseUsingGrobid.setAccessible(true);
+      assertEquals(example1AsBibtex, parseUsingGrobid.invoke(grobidCitationFetcher, example1));
+      assertEquals(example2AsBibtex, parseUsingGrobid.invoke(grobidCitationFetcher, example2));
+      assertEquals(example3AsBibtex, parseUsingGrobid.invoke(grobidCitationFetcher, example3));
+      assertEquals(example4AsBibtex, parseUsingGrobid.invoke(grobidCitationFetcher, example4));
+    }
+
+    /**
+     * Giving an invalid string as a request to the grobid parser, should fail and return an
+     * empty bibtex String.
+     */
+    @Test
+    public void failingGrobidRequestTest() throws IllegalAccessException, InvocationTargetException, NoSuchMethodException {
+        Method parseUsingGrobid = GrobidCitationFetcher.class.getDeclaredMethod("parseUsingGrobid", String.class);
+        parseUsingGrobid.setAccessible(true);
+        assertEquals(defaultReturnValue, parseUsingGrobid.invoke(grobidCitationFetcher, invalidInput1));
+        assertEquals(defaultReturnValue, parseUsingGrobid.invoke(grobidCitationFetcher, invalidInput2));
+    }
+
+    /**
+     * Tests if the Optional<BibEntry> is filled correctly with the extracted fields from bibtex string..
+     */
+    @Test
+    public void grobidParseBibToBibEntryTest() throws IllegalAccessException, InvocationTargetException, NoSuchMethodException {
+        Method parseBibToBibEntry = GrobidCitationFetcher.class.getDeclaredMethod("parseBibToBibEntry", String.class);
+        parseBibToBibEntry.setAccessible(true);
+        Optional<BibEntry> bibEntry = (Optional<BibEntry>)parseBibToBibEntry.invoke(grobidCitationFetcher, example1AsBibtex);
+        assertTrue(bibEntry.isPresent());
+        assertEquals(bibEntry.get().getField(StandardField.AUTHOR).get(), "T Derwing and M Rossiter and M Munro");
+        assertEquals(bibEntry.get().getField(StandardField.TITLE).get(), "Teaching native speakers to listen to foreign-accented speech");
+        assertEquals(bibEntry.get().getField(StandardField.JOURNAL).get(), "Journal of Multilingual and Multicultural Development");
+        assertEquals(bibEntry.get().getField(StandardField.YEAR).get(), "2002");
+    }
+
+    /**
+     * Checks if parseBibToBibEntry creates a empty BibEntry on empty response
+     */
+    @Test
+    public void grobidParseBibToBibEntryReturnsEmptyOptionalWhenFailedTest() throws IllegalAccessException, NoSuchMethodException, InvocationTargetException {
+        Method parseBibToBibEntry = GrobidCitationFetcher.class.getDeclaredMethod("parseBibToBibEntry", String.class);
+        parseBibToBibEntry.setAccessible(true);
+        Optional<BibEntry> bibEntry = (Optional<BibEntry>)parseBibToBibEntry.invoke(grobidCitationFetcher, defaultReturnValue);
+        assertTrue(bibEntry.isPresent());
```

Comment:
Please don't check for BibTeX like this. Use Optional.of(expectedEntry) as expected thing. The rest is done by the assertion magic.